packages/zen-bull-vault/test/e2e/ShutdownEmergencyWithdrawTest.t.sol

```diff
@@ -0,0 +1,257 @@
+
+        uint256 ethIndexPrice =
+            IOracle(ORACLE).getTwap(ETH_USDC_POOL, WETH, USDC, TWAP_PERIOD, true).div(INDEX_SCALE);
+
+        uint256 wPowerPerpToProvide =
+            initialZenBullCrabBalance.wmul(wPowerPerpInCrab).wdiv(IERC20(CRAB).totalSupply());
+        uint256 wethToCaller = wPowerPerpToProvide.wmul(ethIndexPrice).wmul(
+            IController(CONTROLLER).getExpectedNormalizationFactor()
+        );
+
+        // use deal() to get some oSQTH for testing
+        deal(address(OSQTH), address(OWNER), wPowerPerpToProvide);
+
+        // get initial balances
+        uint256 initialOwnerWethBalance = IERC20(WETH).balanceOf(OWNER);
+        uint256 initialOwnerWPowerPerpBalance = IERC20(OSQTH).balanceOf(OWNER);
+
+        // execute emergency withdrawal
+        vm.startPrank(OWNER);
+        IERC20(OSQTH).approve(address(shutdownEmergencyWithdraw), wPowerPerpToProvide);
+        shutdownEmergencyWithdraw.shutdownEmergencyWithdraw();
+        vm.stopPrank();
+
+        // verify the results
+        uint256 finalOwnerWethBalance = IERC20(WETH).balanceOf(OWNER);
+        uint256 finalZenBullCrabBalance = ZenBullStrategy(ZEN_BULL).getCrabBalance();
+        uint256 finalZenBullCrabBalanceFromCrab = IERC20(CRAB).balanceOf(address(ZEN_BULL));
+        uint256 finalOwnerWPowerPerpBalance = IERC20(OSQTH).balanceOf(OWNER);
+
+        // check that ETH was received by deployer
+        assertEq(
+            finalOwnerWethBalance,
+            initialOwnerWethBalance + wethToCaller,
+            "Owner should have received correct WETH"
+        );
+
+        // check that contract has correct eth balance
+        assertEq(
+            IERC20(WETH).balanceOf(address(shutdownEmergencyWithdraw)),
+            wethToReceive - wethToCaller,
+            "ShutdownEmergencyWithdraw should have correct WETH balance"
+        );
+        assertApproxEqRel(
+            IERC20(WETH).balanceOf(address(shutdownEmergencyWithdraw)),
+            wethToCaller,
+            7e16,
+            "The value of WETH should be close to 50% / 50%, within 7%"
+        );
+        // check that Crab balance was reduced and wpowerperp was provided correctly
+        assertEq(finalZenBullCrabBalance, 0, "All Crab should have been redeemed");
+        assertEq(finalZenBullCrabBalanceFromCrab, 0, "All Crab should have been redeemed");
+        assertEq(
+            finalOwnerWPowerPerpBalance,
+            initialOwnerWPowerPerpBalance - wPowerPerpToProvide,
+            "Owner should have provided correct WPowerPerp"
+        );
+    }
+
+    function testClaimZenBullRedemption() public {
+        // get initial crab balance in zen bull strategy
+        uint256 initialZenBullCrabBalance = ZenBullStrategy(ZEN_BULL).getCrabBalance();
+
+        (uint256 ethInCrab, uint256 wPowerPerpInCrab) =
+            ZenBullStrategy(ZEN_BULL).getCrabVaultDetails();
+
+        uint256 ethShare = initialZenBullCrabBalance.wdiv(IERC20(CRAB).totalSupply());
```

Comment:
same comment, more like crabShare but doesn't affect anything

Reply:
changed to share